array_api_compat/common/_helpers.py

```diff
@@ -0,0 +1,175 @@
+"""
+Various helper functions which are not part of the spec.
+
+Functions which start with an underscore are for internal use only but helpers
+that are in __all__ are intended as additional helper functions for use by end
+users of the compat library.
+"""
+from __future__ import annotations
+
+import sys
+
+def _is_numpy_array(x):
+    # Avoid importing NumPy if it isn't already
+    if 'numpy' not in sys.modules:
+        return False
+
+    import numpy as np
+
+    # TODO: Should we reject ndarray subclasses?
```

Comment:
Always a question mark. With "reject", do you mean "convert to ndarray", or "raise", or "ignore", or ....?

Reply:
My thinking here was specifically for `get_namespace`, although this helper is also used in `asarray`. Right now it does accept them, because it uses `isinstance`. I've no idea if an ndarray subclass could implement behavior that would break things here. Presumably if they are properly Liskov substitutable, they shouldn't. My guess is that at worst some functions might return ndarray instead of the appropriate subclass, because we don't properly account for subclases when creating new array objects.

But I haven't tested it. It's possible some people one of the units libraries (I forget which ones subclass ndarray) might care about this at some point, though.

Reply:
> Presumably if they are properly Liskov substitutable, they shouldn't.

`np.matrix` for one isn't, that's a major issue. IIRC also `np.ma.MaskedArray` isn't.